### ISSUE TITLE
Clarification on Prompt Token Placement in vpt_backbone.py'

### ISSUE THREAD

**Issue by SehunChang** [yeyeyeye33/SEC-Prompt#3]
Hello,

Thank you for releasing the code for your excellent paper, "SEC-Prompt: SEmantic Complementary Prompting for Few-Shot Class-Incremental Learning."

I've been studying your vpt_backbone.py file. In the forward_features method (lines 116-120), I noticed the prompt tokens (e.g., TSP, RSP) are concatenated to the end of the input sequence x after the class and patch tokens.

# From vpt_backbone.py
x = torch.cat([x, TSP], dim=1)
...
x = torch.cat([x, RSP], dim=1)
==============

However, the architecture diagrams in the paper (e.g., Figure X) seem to visually represent the prompts as being inserted immediately after the class token.

Could you please clarify the following:

1. Is my interpretation of the code correct? Does the implementation place the prompt tokens at the end of the sequence, not right after the class token?

2. If so, what is the reason for this difference between the code implementation and the visual representation in the paper? Was this a design choice for implementation efficiency, or did you find this placement to yield better performance?

Any clarification on this would be greatly appreciated. Thank you for your time!

**Comment by yeyeyeye33** [comment]
Thank you very much for reading our paper and for your careful examination of the code.

1. Your interpretation is correct. In the released implementation the prompt tokens are appended to the end of the sequence, i.e. after the class token and all patch tokens. In the paper we placed them right after the class token only for the convenience of drawing.

2. The main reason for appending the prompts is to avoid disturbing the original ordering of image patch tokens. Although spatial relations are mostly encoded through the positional embeddings, we preferred to keep the patch sequence untouched because the pretrained ViT may already exploit certain ordering patterns. We did not conduct a systematic study on the influence of the insertion position, so feel free to experiment with other placements—your idea is interesting and might reveal further performance differences.

Thank you again for your attention and valuable questions.
